src/Control/HTTPRequestBuilder.php

```diff
@@ -17,10 +17,13 @@ public static function createFromEnvironment()
     {
         // Clean and update live global variables
         $variables = static::cleanEnvironment(Environment::getVariables());
-        Environment::setVariables($variables); // Currently necessary for SSViewer, etc to work
 
         // Health-check prior to creating environment
-        return static::createFromVariables($variables, @file_get_contents('php://input'));
+        $req = static::createFromVariables($variables, @file_get_contents('php://input'));
+
+        Environment::setVariables($variables); // Currently necessary for SSViewer, etc to work
```

Comment:
Not sure why this order had to be switched. createFromVariables() should not be modifying $variables...